openff/evaluator/protocols/reweighting.py

```diff
@@ -491,11 +492,19 @@ def _reweight_observables(
             observable_dimensions = observable.value.shape[1]
             assert observable_dimensions == 1
 
-            results = mbar.computeExpectations(
-                observable.value.T.magnitude,
-                target_reduced_potentials.value.T.magnitude,
-                state_dependent=True,
-            )
+            if _PYMBAR_MAJOR_VERSION == 3:
+
+                results = mbar.computeExpectations(
+                    observable.value.T.magnitude,
+                    target_reduced_potentials.value.T.magnitude,
+                    state_dependent=True,
+                )
+            elif _PYMBAR_MAJOR_VERSION == 4:
+                results = mbar.compute_expectations(
+                    observable.value.T.magnitude,
+                    target_reduced_potentials.value.T.magnitude,
+                    state_dependent=True,
+                )
```

Comment:
Would it be possible to do the same as `detect_equilibration` and then we can just import from `openff.evaluator.utils.pymbar`?

Reply:
Probably, I'm just not sure about the complexities of soft importing a method vs. a function

Reply:
How I imagine it would work is passing method names as strings and using `getattr`; I don't imagine the methods can be passed alone since they are instance-level and not class-level

```
# pymbar.py
...

    if name == "compute_expectations":
        if _PYMBAR_MAJOR_VERSION == 3:
            return "computeExpectations"
        elif _PYMBAR_MAJOR_VERSION == 4:
            return "compute_expectations"
...

# reweighting.py
...
        from openff.evaluator.utils.pymbar import compute_expectations

            results = getattr(mbar, compute_expectations)(
                observable.value.T.magnitude,
                target_reduced_potentials.value.T.magnitude,
                state_dependent=True,
            )
...
```

Okay, the code isn't as ugly as I imagined it. The readability is okay and it successfully tucks away the logic like the other import. What do you think?

Reply:
I think this is awesome 👍, it makes the code more compact.